source/cellular_bg96_api.c

```diff
@@ -2223,6 +2321,39 @@ static CellularPktStatus_t socketSendDataPrefix( void * pCallbackContext,
     return pktStatus;
 }
 
+static CellularPktStatus_t fileUploadInFileSystem( void * pCallbackContext,
+                                                 char * pLine,
+                                                 uint32_t * pBytesRead )
+{
+    CellularPktStatus_t pktStatus = CELLULAR_PKT_STATUS_OK;
+
+    if( ( pLine == NULL ) || ( pBytesRead == NULL ) )
+    {
+        LogError( ( "socketSendDataPrefix: pLine is invalid or pBytesRead is invalid" ) );
+        pktStatus = CELLULAR_PKT_STATUS_BAD_PARAM;
+    }
+    else if( pCallbackContext != NULL )
+    {
+        LogError( ( "socketSendDataPrefix: pCallbackContext is not NULL" ) );
+        pktStatus = CELLULAR_PKT_STATUS_BAD_PARAM;
+    }
+    else if( *pBytesRead != 2U )

```

Comment:
consider also checking that `pLine[0] == '>'`

Also, an error be returned in case this if is matched

Reply:
pLine[0] == '>' is only on the QISEND / QSSLSEND. In  fileUpload pLine[0] is 'CONNECT', and it is being compared below